### PR TITLE
Mv transfer function from VaultExtension to Vault

### LINKED ISSUE #531: Batch router: BPT transfers and approvals
We're calling `bpt.approve` and `bpt.transfer` in some instances in the batch router. This is kinda inefficient because BPTs are actually facades that call the vault, which in fact calls the extension. So the chain is router --> BPT --> vault --> vault extension.

Can we move the external methods for transfer and approve from the extension to the vault, and then call the vault directly instead of using the BPT facade? In that case, the call chain would be router --> vault.

### PR DESCRIPTION
**Pull request by elshan-eth**
# Description

Move `transfer` function from VaultExtension to Vault for saving gas.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [X] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [ ] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #531

### CODE REVIEW

**Review by joaobrunoah — APPROVED** [review]
LGTM!

File: pkg/interfaces/contracts/vault/IVaultMain.sol [hunk 1]
```diff
@@ -127,6 +127,22 @@ interface IVaultMain {
...
+     * @notice Transfers pool token from owner to a recipient.
+     * @dev Notice that the pool token address is not included in the params. This function is exclusively called by
+     * the pool contract, so msg.sender is used as the token address.
+     *
+     * @param owner Address of the owner
+     * @param to Address of the recipient
+     * @param amount Amount of tokens to transfer
+     * @return success True if successful, false otherwise
+     */
+    function transfer(address owner, address to, uint256 amount) external returns (bool);
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/vault/IVaultMain.sol:144]: Per issue #531, do we also need to move approve?

Comment (jubeira) [reply pkg/interfaces/contracts/vault/IVaultMain.sol:144]: Not really. We're not using that anymore in the batch router.

**Review by jubeira — APPROVED** [review]
LGTM, but I've hijacked the PR.
@EndymionJkb can you approve pls?

**Review by danielmkm — APPROVED** [review]
This looks good

**Pull request merged**